### PR TITLE
Integrated inline images with responsive images (srcset, sizes)

### LINKED ISSUE #8777: Integrate inline images with responsive images (srcset, sizes)

### PR DESCRIPTION
**Pull request by pkwasnik**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (image): Integrated inline images with responsive images. Closes #8777.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-image/tests/imageresize/imageresizeediting.js [hunk 1]
```diff
@@ -136,6 +136,67 @@ describe( 'ImageResizeEditing', () => {
...
+			);
+
+			expect( editor.model.document.getRoot().getChild( 0 ).getChild( 1 ).getAttribute( 'width' ) ).to.equal( '100px' );
+		} );
+
+		it( 'upcasts 50% width correctly', () => {
+			editor.setData( `<p>Lorem <span class="image-inline" style="width:50%;"><img src="${ IMAGE_SRC_FIXTURE }"></span> ipsum</p>` );
+
+			expect( editor.model.document.getRoot().getChild( 0 ).getChild( 1 ).getAttribute( 'width' ) ).to.equal( '50%' );
+		} );
```
Comment (oleq) [inline packages/ckeditor5-image/tests/imageresize/imageresizeediting.js:156]: What about upcasting just ``? Inline images don't exist in the editor data pipeline as `<span></span>` – this is an editing pipeline representation.

Comment (oleq) [reply packages/ckeditor5-image/tests/imageresize/imageresizeediting.js:156]: Even the next text confirms this :point\_down:

File: packages/ckeditor5-image/src/imageresize/imageresizeediting.js [hunk 2]
```diff
@@ -120,7 +122,7 @@ export default class ImageResizeEditing extends Plugin {
 		editor.conversion.for( 'upcast' )
 			.attributeToAttribute( {
 				view: {
-					name: 'figure',
+					name: imageType === 'image' ? 'figure' : 'span',
```
Comment (oleq) [inline packages/ckeditor5-image/src/imageresize/imageresizeediting.js:125]: This conversion is incorrect. In the case of inline images, the upcasted data will be `` but not `<span></span>` (see my comment in tests).

**Review by oleq — CHANGES_REQUESTED** [review]
The feature is incomplete. There's no support for resizing handles 
image.

Also, the upcast conversion pipeline configuration is incorrect. You can check it by setting a size for an inline image in a manual test and doing `editor.setData( editor.getData() );`

**Comment by magda-chrzescian** [comment]
@oleq Resizing the images with the `srcset` causes content jumping in chrome (for both block and inline images):

resizing
---
The inline images are being resized only horizontally in Safari:
Resizing

**Comment by oleq** [comment]
I addressed the Safari issue in [fcd1cfc](https://github.com/ckeditor/ckeditor5/pull/9043/commits/fcd1cfc3a58f39b283068049df4a98c44a26b511). I couldn't reproduce the latter, though. Any ideas?

**Review by magda-chrzescian — APPROVED** [review]
As in the comment: https://github.com/ckeditor/ckeditor5/pull/9043#issuecomment-799445766

**Comment by magda-chrzescian** [comment]
> I addressed the Safari issue in [fcd1cfc](https://github.com/ckeditor/ckeditor5/pull/9043/commits/fcd1cfc3a58f39b283068049df4a98c44a26b511). I couldn't reproduce the latter, though. Any ideas?

I can't reproduce it either, but it wasn't a single case while I was testing. I guess we can merge anyway and keep an eye on it in the inline-images branch and while the QA testing.

**Pull request merged**